src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -43,21 +44,29 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     }
   }
 
-  private def writeStream(dataFrameName: String, writerConfig: StreamingWritingConfiguration,
-                          streamingConfig: Option[Streaming]): Unit = {
+  private def writeStream(dataFrameName: String,
+                          writerConfig: StreamingWritingConfiguration,
+                          streamingConfig: Option[Streaming],
+                          instrumentationProvider: InstrumentationProvider): Unit = {
     log.info(s"Starting to write streaming results of ${dataFrameName}")
-
+    val currentTime = System.currentTimeMillis
```

Comment:
You need to put this inside the function, not before running.
now should be the time the writing ends